### PR TITLE
synquacer: Add DeveloperBox platform support

### PR DESCRIPTION
**Pull request by b49020**
Signed-off-by: Sumit Garg <sumit.garg@linaro.org>

### CODE REVIEW

File: core/arch/arm/plat-synquacer/platform_config.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2018, Linaro Limited
+ * SPDX-License-Identifier: BSD-2-Clause
```
Comment (jforissier) [inline core/arch/arm/plat-synquacer/platform_config.h]: Move to first line: `/* SPDX-License-Identifier: BSD-2-Clause */`

File: core/arch/arm/plat-synquacer/conf.mk [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+CFG_SHMEM_START ?= 0xffc00000
+CFG_SHMEM_SIZE ?= 0x00400000
+endif
+
+ifeq ($(platform-flavor-armv8),1)
+$(call force,CFG_WITH_ARM_TRUSTED_FW,y)
+endif
+
+$(call force,CFG_WITH_LPAE,y)
+ta-targets = ta_arm64
```
Comment (jforissier) [inline core/arch/arm/plat-synquacer/conf.mk]: I'm puzzled here. You're not adding `ta_arm32` to `ta-targets` so this means only 64-bit TAs are supported. Which implies a 64-bit TEE core (kernel), too. But I don't see `CFG_ARM64_core` being set to `y`. 

In order to support "the usual mix" (32-bit kernel with 32-bit TAs, and 64-bit kernel with 32-bit or 64-bit TAs), you would need something like that: https://github.com/OP-TEE/optee_os/blob/master/core/arch/arm/plat-hikey/conf.mk#L13-L20

File: core/arch/arm/plat-synquacer/conf.mk [hunk 3]
```diff
@@ -0,0 +1,28 @@
+PLATFORM_FLAVOR ?= developerbox
+
+$(call force,CFG_GENERIC_BOOT,y)
+$(call force,CFG_SECURE_TIME_SOURCE_CNTPCT,y)
+$(call force,CFG_GIC,y)
+$(call force,CFG_PL011,y)
+$(call force,CFG_PM_STUBS,y)
+
+ifeq ($(PLATFORM_FLAVOR),developerbox)
```
[the remark this answers is not in the document]
Comment (b49020) [reply core/arch/arm/plat-synquacer/conf.mk]: Sure.

File: core/arch/arm/plat-synquacer/conf.mk [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+include core/arch/arm/cpu/cortex-armv8-0.mk
+CFG_TEE_CORE_NB_CORE = 24
```
[the remark this answers is not in the document]
Comment (b49020) [reply core/arch/arm/plat-synquacer/conf.mk]: ok

File: core/arch/arm/plat-synquacer/conf.mk [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+CFG_TZDRAM_START ?= 0xfc000000
+CFG_TZDRAM_SIZE ?= 0x03c00000
+CFG_SHMEM_START ?= 0xffc00000
+CFG_SHMEM_SIZE ?= 0x00400000
+endif
+
+ifeq ($(platform-flavor-armv8),1)
```
[the remark this answers is not in the document]
Comment (b49020) [reply core/arch/arm/plat-synquacer/conf.mk]: ok

File: core/arch/arm/plat-synquacer/conf.mk (hunk 2 above)
Comment (b49020) [reply core/arch/arm/plat-synquacer/conf.mk]: I did pass CFG_ARM64_core=y from cmdline. Let me add $(call force,CFG_ARM64_core,y) in this file instead. For now I target only 64-bit TAs. In future, if we see any requirement for 32-bit TAs, we can add corresponding target.

File: core/arch/arm/plat-synquacer/conf.mk [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+$(call force,CFG_WITH_ARM_TRUSTED_FW,y)
+endif
+
+$(call force,CFG_WITH_LPAE,y)
+ta-targets = ta_arm64
+
+CFG_CRYPTO_SIZE_OPTIMIZATION ?= n
+CFG_WITH_STACK_CANARIES ?= y
+CFG_ARM_GICV3 := y
+CFG_CORE_CLUSTER_SHIFT := 1
```
[the remark this answers is not in the document]
Comment (b49020) [reply core/arch/arm/plat-synquacer/conf.mk]: ok

File: core/arch/arm/plat-synquacer/main.c [hunk 7]
```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2018, Linaro Limited
+ * SPDX-License-Identifier: BSD-2-Clause
```
[the remark this answers is not in the document]
Comment (b49020) [reply core/arch/arm/plat-synquacer/main.c]: ok

File: core/arch/arm/plat-synquacer/main.c [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+ */
+
+#include <platform_config.h>
```
[the remark this answers is not in the document]
Comment (b49020) [reply core/arch/arm/plat-synquacer/main.c]: ok

File: core/arch/arm/plat-synquacer/conf.mk (hunk 2 above)
Comment (jforissier) [reply core/arch/arm/plat-synquacer/conf.mk]: Fine with me.

File: core/arch/arm/plat-synquacer/platform_config.h (hunk 1 above)
Comment (b49020) [reply core/arch/arm/plat-synquacer/platform_config.h]: Shall I add `// SPDX-License-Identifier: BSD-2-Clause` instead in first line?

Comment (jforissier) [reply core/arch/arm/plat-synquacer/platform_config.h]: This is a header file so `/* SPDX-License-Identifier: BSD-2-Clause */` (C-style comment) in first line please

Comment (b49020) [reply core/arch/arm/plat-synquacer/platform_config.h]: ok

**Comment by b49020** [comment]
@jforissier Updated PR to incorporate comments. Please review again.

**Comment by jforissier** [comment]
Looks good, thanks.

`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Pull request merged**